src/QueryState.js

```diff
@@ -54,33 +50,19 @@ export class QueryState {
     return {unsubscribe};
   }
 
-  _runStaticQuery(query, runQuery) {
-    const promise = runQuery(query);
-    this.closeHandlers.push(() => promise.then(x => isCursor(x) && x.close()));
-    promise.then(cursor => {
-      if (isCursor(cursor)) {
-        cursor.toArray().then(result => {
-          this._updateValue(result);
-        });
-      } else {
-        this._updateValue(cursor);
-      }
-    }, error => {
-      this._addError(error);
-    });
-  }
-
-  _runChangeQuery(query, runQuery) {
+  _runQuery(query, runQuery, changes = false) {
     const changeQuery = query.changes({includeStates: true, includeInitial: true});
-    const promise = runQuery(changeQuery);
+    const promise = runQuery(changes ? changeQuery : query);
     this.closeHandlers.push(() => promise.then(x => isCursor(x) && x.close()));
     promise.then(cursor => {
+      const isFeed = !!cursor.constructor.name.match(/Feed$/);
       const isPointFeed = cursor.constructor.name === 'AtomFeed';
-      this.value = isPointFeed ? undefined : [];
-      cursor.each((error, row) => {
-        if (error) {
-          this._addError(error);
-        } else {
+      if (isFeed) {
+        this.value = isPointFeed ? undefined : [];
+        cursor.each((error, row) => {
+          if (error) {
+            return this._addError(error);
```

Comment:
Any reason not to keep the original logic of putting the code below in the `else` block and not having a `return` here?

I'm guessing it's probably the same behavior, but safer to change less, and I think using `return` for a value that's always undefined is a bit misleading.